CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 3.12)
+project(opus LANGUAGES C)
+include(opus_sources.cmake)
+enable_testing()
+
+message(STATUS "Build target:")
+message(STATUS "  CMAKE_SYSTEM_NAME: ${CMAKE_SYSTEM_NAME}")
+message(STATUS "  CMAKE_SYSTEM_PROCESSOR: ${CMAKE_SYSTEM_PROCESSOR}")
+message(STATUS "  CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+
+message(STATUS "Builds on:")
+message(STATUS "  CMAKE_HOST_SYSTEM_NAME: ${CMAKE_HOST_SYSTEM_NAME}")
+message(STATUS "  CMAKE_HOST_SYSTEM_VERSION: ${CMAKE_HOST_SYSTEM_VERSION}")
+message(STATUS "  CMAKE_HOST_SYSTEM_PROCESSOR: ${CMAKE_HOST_SYSTEM_PROCESSOR}")
+
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+  message(FATAL_ERROR "Only tested on Windows and Linux")
+endif()
+
+list(APPEND opus_sources
+            ${opus_sources_float}
+            ${celt_sources}
+            ${silk_sources})
+
+# TODO: add Fixed and optimizations
+list(APPEND opus_sources ${silk_sources_float})
+
+# library
+add_library(opus ${opus_sources})
+set_property(TARGET opus PROPERTY C_STANDARD 99)
+target_include_directories(opus
+                           PUBLIC $<BUILD_INTERFACE:
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/include
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/celt
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/silk
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/silk/float
+                                  >
+                                  $<INSTALL_INTERFACE:include>)
+target_compile_definitions(opus PRIVATE USE_ALLOCA CUSTOM_MODES OPUS_BUILD)
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
```

Comment:
The ${} expansion around `CMAKE_SYSTEM_NAME` is not needed and could trigger double-expansion.
```cmake
if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
will do the right thing :)

Reply:
right thanks!